File: src/main/js/college.js
function Student(id, name) {
    this.lang = "JS"
    this.id = id;
    this.name = name;
}

Student.prototype.setId = function(id) {
    this.id = id;
}

Student.prototype.setName = function(name) {
    this.name = name;
}

Student.prototype.getID = function() {
    return this.id;
}

Student.prototype.getName =function() {
    return this.name;
}


class CSEStudent extends Student {
    constructor(id, name){
        super(id, name)
    }

    getBranch(){
        return "CSE"
    }

    getLang(){
        return this.lang
    }
}

class College {
    constructor(name) {
        this.name = name;
        this.students = [];
        this.students.constructor = function constructor() {
            return [];
        }
    }

    getName() {
        return this.name;
    }
    
    addStudent(student) {
        this.students.push(student);
    }

    getStudents() {
        return this.students;
    }
}


//let college = new College("test");
//let cseStudent = new CSEStudent();

//college.addStudent(cseStudent)

//console.log(college.getStudents())
//module.exports = College;

File: src/main/ruby/college.rb
#!/usr/bin/ruby

$LOAD_PATH << '/home/paas/DCAP/DCAP-Sapphire/multilanguage/graalvm_poc/src/main/ruby'

require 'student'
require 'csestudent'

class College
    def initialize(name)
        @name = name
        @students = []
    end

    def getName()
        return @name
    end

    def addStudent(student)
        @students.push(student)
    end

    def getStudents()
        return @students
    end
end